core/src/main/java/org/aerogear/mobile/core/MobileCore.java

```diff
@@ -88,15 +96,23 @@ public void bootstrap() {
     }
 
     /**
-     *
+     * There are some services that are "core" and usually won't appear in mobile-services.json
+     */
+    private void addCoreServices() {
```

Comment:
I think that core services may be well known at compilation time and will not change. 
Having runtime, string based helpers (I do not want to call them services to not make confusion with top level SDK) seems like we trying to invent something that is already supported by Java language. I might be missing actual reason, but we could just make core helpers available as part of the API. WDYT?  Service configuration is not needed here and it may be misleading. 

I do not have strong opinion about it, but this code definitely sticks out of the rest of the changes and just wanted to point that out.

Reply:
@wtrocki You are probably right.  So what I was trying to do was 1) test the servicemodule concept with as much usage as I could and 2) provide a mechanism for hiding concrete classes.  The SMR just really isn't the right way is the feeling I am getting.

I'll take it to the mailing soon, now that we've gotten more ideas about other parts of the system it is a perfect time to revisit.